### PR TITLE
Support unique data tests on radio field group

### PR DESCRIPTION
**Pull request by carsoli**

### CODE REVIEW

File: packages/documentation/pages/usage/components/form.vue [hunk 1]
```diff
@@ -323,9 +324,9 @@ export default defineComponent({
 				/* eslint-enable no-magic-numbers */
 			}),
 			genderOptions: computed((): Kotti.FieldRadioGroup.Props['options'] => [
-				{ label: 'MALE', value: 'male' },
+				{ label: 'MALE', value: 'male', dataTest: 'unique-male-data-test' },
 				{ label: 'FEMALE', value: 'female' },
```
Comment (santiagoballadares) [inline packages/documentation/pages/usage/components/form.vue:328]: Is it intended not to give dataTest to this option?

File: packages/kotti-ui/source/kotti-field-radio-group/KtFieldRadioGroup.vue [hunk 2]
```diff
@@ -96,6 +98,15 @@ export default defineComponent<KottiFieldRadioGroup.PropsInternal>({
...
 				forceUpdate()
 			},
+			optionDataTest: (option: KottiFieldRadioGroup.Entry) => {
+				if (option.dataTest) return option.dataTest
+
+				if (Object.keys(field.inputProps).includes('data-test')) {
+					return [field.inputProps['data-test'], option.value].join('.')
+				}
+
+				return undefined
```
Comment (santiagoballadares) [inline packages/kotti-ui/source/kotti-field-radio-group/KtFieldRadioGroup.vue:108]: Is it necessary this return? Not returning anything from a function will default to undefined

**Review by HansAuger — APPROVED** [review]

File: packages/documentation/pages/usage/components/form.vue (hunk 1 above)
Comment (carsoli) [reply packages/documentation/pages/usage/components/form.vue:328]: yes so i was basically trying to see that i actually support data tests per option (which is respected here)

and if i don't pass it , i checked in the docs that the rendered data-test was 
`field.dataTest.${optionValue}`

File: packages/kotti-ui/source/kotti-field-radio-group/KtFieldRadioGroup.vue (hunk 2 above)
Comment (carsoli) [reply packages/kotti-ui/source/kotti-field-radio-group/KtFieldRadioGroup.vue:108]: you're right, just wanted to make it explicit that the data-test would be unbound on html

**Review by santiagoballadares — DISMISSED** [review]

**Review by RueRivoli — DISMISSED** [review]

**Review by FlorianWendelborn — CHANGES_REQUESTED** [review]

File: packages/documentation/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
 	"dependencies": {
-		"@3yourmind/kotti-ui": "3.0.0-beta.27",
```
[the remark this answers is not in the document]
Comment (carsoli) [reply packages/documentation/package.json:3]: ye it's currently being resolved by me

File: packages/documentation/pages/usage/components/form.vue [hunk 4]
```diff
@@ -263,6 +263,7 @@
 						</KtFormControllerList>
 					</ul>
 					<KtFieldRadioGroup
+						dataTest="unique-radio-group"
```
[the remark this answers is not in the document]
Comment (carsoli) [reply packages/documentation/pages/usage/components/form.vue:266]: yes it's to test the dataTest when it's passed on the component level and on options

File: packages/kotti-ui/source/kotti-field-radio-group/KtFieldRadioGroup.vue [hunk 5]
```diff
@@ -85,9 +86,10 @@ export default defineComponent<KottiFieldRadioGroup.PropsInternal>({
 			forceUpdateKey: forceUpdateKey.value,
 			inputProps: computed(
 				(): Partial<HTMLInputElement & { class: string }> => ({
-					...field.inputProps,
```
[the remark this answers is not in the document]
Comment (carsoli) [reply packages/kotti-ui/source/kotti-field-radio-group/KtFieldRadioGroup.vue:88]: the only other way i'd do this is to omit the dataTest from inputProps

idk if you fully get what this is doing right now

[the remark this answers is not in the document]
Comment (FlorianWendelborn) [reply packages/kotti-ui/source/kotti-field-radio-group/KtFieldRadioGroup.vue:88]: I guess you could use `..._.omit(inputProps, ['dataTest'])` instead then

**Comment by HansAuger** [comment]
I took over because Carol is sick but this urgently needed for having passing e2e tests

**Comment by carsoli** [comment]
squashing and merging :)

**Pull request merged**